tests/lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,435 @@
+        }
+      }`,
+      output: `
+      import { inject as service } from '@ember/service';
+      import Controller from '@ember/controller';
+      import { action } from '@ember/object';
+
+      export default class NewPostController2 extends Controller {
+        @service('router') routerService;
+
+        @action
+        async save({ title, text }) {
+          let post = this.store.createRecord('post', { title, text });
+          await post.save();
+          return this.routerService.transitionTo('post', post.id);
+        }
+      }`,
+      errors: [
+        {
+          messageId: 'main',
+          type: 'MemberExpression',
+        },
+      ],
+    },
+    {
+      filename: 'controllers/new-post.js',
+      code: `
+      import { inject as serviceInjection } from '@ember/service';
+      import Controller from '@ember/controller';
+      import { action } from '@ember/object';
+
+      export default class NewPostController3 extends Controller {
+        @serviceInjection('router') routerService;
+
+        @action
+        async save({ title, text }) {
+          let post = this.store.createRecord('post', { title, text });
+          await post.save();
+          return this.transitionToRoute('post', post.id);
+        }
+      }`,
+      output: `
+      import { inject as serviceInjection } from '@ember/service';
+      import Controller from '@ember/controller';
+      import { action } from '@ember/object';
+
+      export default class NewPostController3 extends Controller {
+        @serviceInjection('router') routerService;
+
+        @action
+        async save({ title, text }) {
+          let post = this.store.createRecord('post', { title, text });
+          await post.save();
+          return this.routerService.transitionTo('post', post.id);
+        }
+      }`,
+      errors: [
+        {
+          messageId: 'main',
+          type: 'MemberExpression',
+        },
+      ],
+    },
+    // Existing injection name other than service
+    {
+      filename: 'routes/index.js',
+      code: `
+      import Route from '@ember/routing/route';
+      import { inject as injectService } from '@ember/service';
+
+      export default class SettingsRoute extends Route {
+        @injectService session;
+
+        beforeModel() {
+          if (!this.session.isAuthenticated) {
+            this.transitionTo('login');
+          }
+        }
+      }`,
+      output: `
+      import Route from '@ember/routing/route';
+      import { inject as injectService } from '@ember/service';
+
+      export default class SettingsRoute extends Route {
+        @injectService('router') router;
+@injectService session;
+
+        beforeModel() {
+          if (!this.session.isAuthenticated) {
+            this.router.transitionTo('login');
+          }
+        }
+      }`,
+      errors: [
+        {
+          messageId: 'main',
+          type: 'MemberExpression',
+        },
+      ],
+    },
+    {
```

Comment:
Since these test cases are pretty bulky, a brief comment in front of each could be helpful to indicate the goal of each test case (what it's trying to test).